tests/ignite/metrics/test_accuracy.py

```diff
@@ -91,145 +91,44 @@ def _test():
-        assert acc._type == "binary"
-        assert isinstance(acc.compute(), float)
-        assert accuracy_score(np_y, np_y_pred) == pytest.approx(acc.compute())
+def test_binary_input():
+    acc = Accuracy()
 
+    def _test(y_pred, y, n_iters):
         acc.reset()
-        y_pred = torch.randint(0, 2, size=(4, 1, 12)).long()
-        y = torch.randint(0, 2, size=(4, 12)).long()
         acc.update((y_pred, y))
         np_y = y.numpy().ravel()
         np_y_pred = y_pred.numpy().ravel()
-        assert acc._type == "binary"
-        assert isinstance(acc.compute(), float)
-        assert accuracy_score(np_y, np_y_pred) == pytest.approx(acc.compute())
 
-        # Batched Updates
-        acc.reset()
-        y_pred = torch.randint(0, 2, size=(100, 1, 8, 8)).long()
-        y = torch.randint(0, 2, size=(100, 8, 8)).long()
-
-        batch_size = 16
-        n_iters = y.shape[0] // batch_size + 1
-
-        for i in range(n_iters):
-            idx = i * batch_size
-            acc.update((y_pred[idx : idx + batch_size], y[idx : idx + batch_size]))
-
-        np_y = y.numpy().ravel()
-        np_y_pred = y_pred.numpy().ravel()
+        if n_iters > 1:
+            # Batched Updates
+            batch_size = y.shape[0] // n_iters + 1
+            for i in range(n_iters):
+                idx = i * batch_size
+                acc.update((y_pred[idx : idx + batch_size], y[idx : idx + batch_size]))
         assert acc._type == "binary"
         assert isinstance(acc.compute(), float)
         assert accuracy_score(np_y, np_y_pred) == pytest.approx(acc.compute())
 
-    # check multiple random inputs as random exact occurencies are rare
-    for _ in range(10):
-        _test()
+    test_cases = [
+        # Binary accuracy on input of shape (N, L)
+        (torch.randint(0, 2, size=(10, 5)).long(), torch.randint(0, 2, size=(10, 5)).long(), 1),
+        (torch.randint(0, 2, size=(10, 1, 5)).long(), torch.randint(0, 2, size=(10, 1, 5)).long(), 1),
+        (torch.randint(0, 2, size=(100, 8)).long(), torch.randint(0, 2, size=(100, 8)).long(), 16),
+        # Binary accuracy on input of shape (N, H, W, ...)
+        (torch.randint(0, 2, size=(4, 1, 12, 10)).long(), torch.randint(0, 2, size=(4, 1, 12, 10)).long(), 1),
+        (torch.randint(0, 2, size=(4, 1, 12, 10)).long(), torch.randint(0, 2, size=(4, 1, 12, 10)).long(), 1),
+        (torch.randint(0, 2, size=(100, 8, 8)).long(), torch.randint(0, 2, size=(100, 8, 8)).long(), 16),
+        # Binary accuracy on input of shape (N, 1, ...) - Multiclass input
+        (torch.randint(0, 2, size=(4, 1)).long(), torch.randint(0, 2, size=(4,)).long(), 1),
+        (torch.randint(0, 2, size=(4, 1, 12)).long(), torch.randint(0, 2, size=(4, 12)).long(), 1),
+        (torch.randint(0, 2, size=(100, 1, 8, 8)).long(), torch.randint(0, 2, size=(100, 8, 8)).long(), 16),
+        # Multiclass input data of shape (N, ) and (N, C)
+    ]
+    for y_pred, y, n_iters in test_cases:
```

Comment:
This is not the best way to generate test as the `test_cases` have been generated already, Please have a look [here](https://github.com/pytorch/ignite/blob/16b3a68b164aa140ee0e78161f8d2d183dea600e/tests/ignite/contrib/metrics/test_roc_auc.py#L90)